src/main/scala/io/iohk/ethereum/ledger/BlockImport.scala

```diff
@@ -110,7 +110,7 @@ class BlockImport(
       block: Block,
       currentBestBlock: Block,
       currentWeight: ChainWeight
-  )(implicit blockExecutionContext: ExecutionContext): Future[BlockImportResult] = Future {
+  )(implicit blockExecutionContext: ExecutionContext): Task[BlockImportResult] = Task.evalOnce {
```

Comment:
Why is this `evalOnce`, or the ones above, needed? Isn't the task only evaluated once?

Reply:
I thought the same. But some tests fail if I just use `Task.eval`, precisely because the mock expectations only expect 1 call. I would like to understand why in the tests this code is evaluated more than once, but this is the best I found for now.